graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -163,44 +167,59 @@ def _get_query_interval_of_integer_inequality_filter(
     return query_interval
 
 
-# TODO(bojanserafimov): This function can be simplified, as len(filter_fields) is always 1.
-def _estimate_filter_selectivity_of_equality(schema_info, location_name, filter_fields):
-    """Calculate the selectivity of equality filter(s) at a given location.
-
-    Using the available unique indexes and/or the distinct_field_values_count statistic, this
-    function extracts the current location's selectivites, and then combines them, returning one
-    Selectivity object.
+def _estimate_filter_selectivity_of_in_collection(
+    schema_info: QueryPlanningSchemaInfo,
+    location_name: str,
+    filter_field: str,
+    collection: Optional[Set[Any]],
+):
+    """Calculate the selectivity of in_collection filter.
 
     Args:
         schema_info: QueryPlanningSchemaInfo
-        location_name: string, type of the location being filtered
-        filter_fields: tuple of str, listing all the fields being filtered over
+        location_name: type name of the location being filtered
+        filter_field: field affected by the filter
+        collection: the values the filter allows, if known. A value of None means the collection
+                    is not known at compile time.
```

Comment:
Overview of changes in this particular function:
- Change `filter_fields` to `filter_field`. The function previously incorrectly assumed that if an index covers the `filter_fields`, then this filter is very selective. This is not the case since `filter_fields` are loosly defined, and can mean anything. In the case of `in_collection` and `=` filters, there's always just one filter field, so the wrong codepath was actually never called.
- Previously this function was a helper for `=` filters that was also used in estimating `in_collection` selectivity. We can't do that anymore since we dropped the assumption that the filter value doesn't matter. We need to look at all values inside the collection. So now we compute `=` filter selectivity using this helper by passing a single-element list for the `collection` argument.